community/import-tool/src/main/java/org/neo4j/tooling/ImportTool.java

```diff
@@ -396,6 +392,22 @@ public static void main( String[] incomingArguments, boolean defaultSettingsSuit
             }
         }
 
+        configuration = importConfiguration( processors, defaultSettingsSuitableForTests, dbConfig, pageSize );
+        input = new CsvInput( nodeData( inputEncoding, nodesFiles ), defaultFormatNodeFileHeader(),
+                relationshipData( inputEncoding, relationshipsFiles ), defaultFormatRelationshipFileHeader(),
+                idType, csvConfiguration( args, defaultSettingsSuitableForTests ), badCollector,
+                configuration.maxNumberOfProcessors() );
+
+        doImport( out, err, storeDir, nodesFiles, relationshipsFiles,
```

Comment:
Disregarding the fact that badOutput is closed before it is used (see previous comment), here badOutput is never closed.

Any Streams should be closed when

1) An error occurs, and we are terminating execution
2) Once there is no more need for the stream (we're done)

And it would typically be the responsibility of whoever opened the stream to then close it again.